src/Streamly/Internal/Data/Stream/IsStream/Expand.hs

```diff
@@ -494,6 +495,18 @@ unfoldManyRoundRobin ::(IsStream t, Monad m)
 unfoldManyRoundRobin u m =
     fromStreamD $ D.unfoldManyRoundRobin u (toStreamD m)
 
+-- XXX This can be modularly expressed as transformation and merging of two
+-- pipes once we have the Pipe abstraction.
+--
+-- | Like 'unfoldMany' but unfolds @Left a@ into @Left b@ and @Right a@ into
+-- @Right b@.
+--
+-- /Pre-release/
+{-# INLINE unfoldManyEither #-}
+unfoldManyEither ::(IsStream t, Monad m) =>
+    Unfold m a b -> t m (Either a a) -> t m (Either b b)
+unfoldManyEither u m = fromStreamD $ D.unfoldManyEither u (toStreamD m)
+
```

Comment:
Wouldn't `unfoldMany` on an unfold of type `Unfold (Either a a) (Either b b)` work?

Reply:
We need a transformation using `Unfold m a b`. If we can transform `Unfold m a b` to `Unfold m (f a) (f b)` using a `f a -> f b` then it might work.